### DISCUSSION TITLE
pip install

### DISCUSSION THREAD

**Discussion by alagerberg** [kineticstoolkit/kineticstoolkit#234] [1. Users Q&A] [answered]
In our institution we use spyder via PIP install. If we install the kinetic toolkit that way we run into problems. the function used for importing c3d files does not seem to be available through PIP. Is there a solution (other than using conda install)

**Comment by felixchenier** [comment]
Hi @alagerberg 

Unfortunately, ezc3d (the software we rely on to read and write C3D files) is not distributed via PIP: see this [https://github.com/pyomeca/ezc3d/issues/111](issue) for more info why.

Therefore there is no easy solution for you. Other than using conda, you can try to [compile ezc3d](https://github.com/pyomeca/ezc3d?tab=readme-ov-file#compiling-for-windows-linux-and-mac). It's a road I've never tried, but I heard successes :-)

Are you a user or a sysadmin? If you're simply a user trying to get things done, just know that installing miniconda doesn't require admin rights: everything can be installed in a user's folder. You could install your own environment there, install kineticstoolkit in this environement, and make Spyder use this environment instead:

![Screen Shot 2024-03-26 at 11 45 19](https://github.com/kineticstoolkit/kineticstoolkit/assets/34967663/72d8db55-4e39-43b2-8754-3a6e77b0da1f)

Or, install everything including Spyder in this user environment. This is the best I can see at this moment.

**Reply by alagerberg** [reply]
Thanks for the quick respons. We will look into the different options

**Reply by alagerberg** [reply]
Hi Felix I am right that there is a new comment on the site about PIP installing?
"Although we don’t cover this installation method in this guide, everything is also available using pip for those who prefer managing their virtual environments manually".
Does this mean that the issue with installing ezc3d with PIP is solved?

**Reply by felixchenier** [reply]
@alagerberg Yes it is! We thank @pariterre (ezc3d creator/maintainer) for getting ezc3d available on pip :-)

**Reply by alagerberg** [reply]
Great. Thanks

**Comment by felixchenier** [comment, marked as answer]
ezc3d is now available with pip, and so is the complete KTK framework :-)
